app/connection_manager.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+from psycopg_pool import ConnectionPool
+
+
+class ConnectionManager:
+    """
+    Manages database connections using a connection pool for PostgreSQL.
+
+    This class is used for managing database connections, transactions, and pooling
+    efficiently. It can be utilized as a context manager, making sure that resources
+    are properly managed (connections are committed, rolled back, and released).
+
+    In testing mode, it avoids committing transactions and releasing connections
+    back to the pool, enabling more control over connections.
+
+    Attributes:
+        testing (bool): Indicates if the application is in testing mode.
+        pool (ConnectionPool): The connection pool for managing database connections.
+        connection (Connection): The active database connection, if any.
+    """
+
+    def __init__(self, pool: ConnectionPool):
+        """
+        Initializes the ConnectionManager with a connection pool.
+
+        Args:
+            pool (ConnectionPool): The connection pool used to manage database connections.
+        """
+        self.testing = os.getenv("TESTING", "") == "True"
+        self.pool = pool
+        self.connection = None
+
```

Comment:
Do we need validation on the input ConnectionPool?

Reply:
Done.